Header header
---
bool begin_grasp